tests/behat/travis-upload-artifacts.php

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env php
+<?php
+/**
+ * Creates an index.html with links to all files in a given directory structure, recursively.
+ * This is useful for Amazon S3 uploads with static file hosting, since it doesn't list files by default.
+ *
+ * Assumes to run in a SilverStripe webroot
+ */
+
+/**
+ * Run a command
+ *
+ * @param string $cmd
+ * @param bool $echo
+ */
+function run($cmd, $echo = true) {
+	if($echo) echo "+ $cmd\n";
+	passthru($cmd, $returnVar);
+	if($returnVar > 0) die($returnVar);
+}
+
+/**
+ * Check if an env variable is set
+ *
+ * @param $envs
+ * @return bool
+ */
+function checkenv($envs) {
+	if($envs) {
+		foreach(explode(',',$envs) as $env) {
+			if(!getenv($env)) {
+				return false;
+			}
+		}
+	}
+	return true;
+}
+
+$opts = getopt('', array(
+	'artifacts-path:',
+	'target-path:',
+	'if-env:',
+	'artifacts-base-url:',
+));
+
+// --if-env=BEHAT_TEST means that this script will only be executed if the given environment var is set
+if (empty($opts['if-env'])) {
+	echo "--if-env option is mandatory";
```

Comment:
Enforced --if-env to prevent this script being invoked outside of the correct context.